pandas/tseries/offsets.py

```diff
@@ -221,6 +232,60 @@ def apply(self, other):
         else:
             return other + timedelta(self.n)
 
+    @apply_index_wraps
+    def apply_index(self, i):
+        """
+        Vectorized apply of DateOffset to DatetimeIndex,
+        raises NotImplentedError for offsets without a
+        vectorized implementation
+
+        Parameters
+        ----------
+        i : DatetimeIndex
+
+        Returns
+        -------
+        y : DatetimeIndex
+        """
+
+        relativedelta_fast = set(['years', 'months', 'weeks',
+                                  'days', 'hours', 'minutes',
+                                  'seconds', 'microseconds'])
+        # relativedelta/_offset path only valid for base DateOffset
+        if type(self) is DateOffset:
```

Comment:
yes I think those should raise ``NotImplementedError`` if they don't support a feature. You may have to an a method to ``DateOffset`` that raises for non-valid sub-classes (trying to think of a good word for those).